samples/nrf9160/modem_shell/src/ppp/ppp_settings.c

```diff
@@ -0,0 +1,227 @@
+			shell_error(shell_global, "Failed to read uart_flow_ctrl, error: %d", ret);
+			return ret;
+		}
+		return 0;
+	}
+	return 0;
+}
+
+/******************************************************************************/
+
+int ppp_uart_settings_read(struct uart_config *uart_conf)
+{
+	if (ppp_uart_conf_valid) {
+		memcpy(uart_conf, &ppp_uart_conf, sizeof(struct uart_config));
+		return 0;
+	} else {
+		return -EINVAL;
+	}
+}
+
+int ppp_uart_settings_write(struct uart_config *uart_conf)
+{
+	int ret;
+	char *key;
+
+	key = PPP_SETT_KEY "/" PPP_SETT_UART_BAUDRATE;
+	ret = settings_save_one(key, &(uart_conf->baudrate), sizeof(uart_conf->baudrate));
+	if (ret) {
+		goto return_err;
+	}
+
+	key = PPP_SETT_KEY "/" PPP_SETT_UART_PARITY;
+	ret = settings_save_one(key, &(uart_conf->parity), sizeof(uart_conf->parity));
+	if (ret) {
+		goto return_err;
+	}
+
+	key = PPP_SETT_KEY "/" PPP_SETT_UART_STOP_BITS;
+	ret = settings_save_one(key, &(uart_conf->stop_bits), sizeof(uart_conf->stop_bits));
+	if (ret) {
+		goto return_err;
+	}
+
+	key = PPP_SETT_KEY "/" PPP_SETT_UART_DATA_BITS;
+	ret = settings_save_one(key, &(uart_conf->data_bits), sizeof(uart_conf->data_bits));
+	if (ret) {
+		goto return_err;
+	}
+
+	key = PPP_SETT_KEY "/" PPP_SETT_UART_FLOW_CTRL;
+	ret = settings_save_one(key, &(uart_conf->flow_ctrl), sizeof(uart_conf->flow_ctrl));
+	if (ret) {
+		goto return_err;
+	}
+
+	ret = ppp_settings_uart_dev_conf_set(uart_conf);
+	if (ret) {
+		shell_error(shell_global, "Failed to set setting to PPP uart %s, error: %d", ret);
+		return ret;
+	}
+
+	memcpy(&ppp_uart_conf, uart_conf, sizeof(ppp_uart_conf));
+	return 0;
+
+return_err:
+	shell_error(shell_global, "Failed to save key %s, error: %d", key, ret);
+	return ret;
+}
+
+/******************************************************************************/
+
+static struct settings_handler cfg = { .name = PPP_SETT_KEY, .h_set = ppp_settings_handler };
+
+int ppp_settings_init(void)
+{
+	int ret = 0;
+
+	ppp_uart_conf_valid = false;
+
+	ppp_uart_dev = device_get_binding(CONFIG_NET_PPP_UART_NAME);
+	if (!ppp_uart_dev) {
+		shell_warn(shell_global, "Cannot get ppp dev binding");
+		ppp_uart_dev = NULL;
+	}
+
+	/* Read defaults: */
+	ret = ppp_settings_uart_dev_conf_get(&ppp_uart_conf);
+	if (ret) {
+		shell_error(shell_global, "Failed to get current uart settings, error: %d", ret);
+		return ret;
+	}
+
+	ret = settings_subsys_init();
+	if (ret) {
+		shell_error(shell_global, "Failed to initialize settings subsystem, error: %d",
+			    ret);
+		return ret;
+	}
+
+	ret = settings_register(&cfg);
+	if (ret) {
+		shell_error(shell_global, "Cannot register settings handler %d", ret);
+		return ret;
+	}
+
+	ret = settings_load();
+	if (ret) {
+		shell_error(shell_global, "Cannot load settings %d", ret);
+		return ret;
+	}
+
+	/* Set possible custom uart config: */
+	ret = ppp_settings_uart_dev_conf_set(&ppp_uart_conf);
+	if (ret) {
+		shell_error(shell_global, "Failed to get current uart settings, error: %d", ret);
```

Comment:
The error text is probably copy pasted from above and should be changed a bit?

Reply:
Fixed.